### PR TITLE
Fix TestRunFinished.success message to return true when the run has passed

### PR DESCRIPTION
**Pull request by aurelien-reeves**
# Description

Fixes. #1604

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

Your PR is ready for review once the following checklist is
complete. You can also add some checks if you want to.

- [x] Tests have been added for any changes to behaviour of the code
- [x] New and existing tests are passing locally and on CI
- [x] `bundle exec rubocop` reports no offenses
- [x] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
Sorry for the messed history. I guess it will be clean once merged

File: features/docs/formatters/message.feature [hunk 1]
```diff
@@ -43,3 +43,28 @@ Feature: Message output formatter
...
+    Given a file named "features/steps.rb" with:
+      """
+      Given('a passed step') {}
+      Given('a failed step') { skip_this_scenario }
+      """
+    When I run `cucumber features/my_feature.feature --format message`
+    Then output should be valid NDJSON
+    And the output should contain:
+    """
+    {"testRunFinished":{"success":true
```
Comment (luke-hill) [inline features/docs/formatters/message.feature:69]: Qu) Is the state only true/false? so pending is classed as true? Could we maybe just make the failed step actually be a passing step just to make me feel better

File: features/docs/formatters/message.feature [hunk 2]
```diff
@@ -43,3 +43,28 @@ Feature: Message output formatter
...
+    """
```
Comment (luke-hill) [inline features/docs/formatters/message.feature:70]: newline

File: lib/cucumber/runtime.rb [hunk 3]
```diff
@@ -79,7 +79,7 @@ def run!
 
       receiver = Test::Runner.new(@configuration.event_bus)
       compile features, receiver, filters, @configuration.event_bus
-      @configuration.notify :test_run_finished
+      @configuration.notify :test_run_finished, !failure?
```
Comment (luke-hill) [inline lib/cucumber/runtime.rb:82]: To avoid setting public in private can we just move this up to above private?

**Review by luke-hill — COMMENTED** [review]
Minor queries

File: features/docs/formatters/message.feature (hunk 1 above)
Comment (aurelien-reeves) [reply features/docs/formatters/message.feature:69]: Success is true or false, yes.
I took the same way the CLI use to exit with 0 or not.
Pending step are passing or failing depending the strict mode.

I can change the test as requested, yes 👌

File: features/docs/formatters/message.feature (hunk 2 above)
Comment (aurelien-reeves) [reply features/docs/formatters/message.feature:70]: Where do you want a newline? Wouldn't you an indent for the docstring intead?

File: lib/cucumber/runtime.rb (hunk 3 above)
Comment (aurelien-reeves) [reply lib/cucumber/runtime.rb:82]: Don't get that comment, sorry 🤔

Comment (luke-hill) [reply lib/cucumber/runtime.rb:82]: This method you're using (I think it's a method), is a hidden public/private method (As in it's actually public, but nestled way down deep inside the private methods, then toggled back to public).

Ideally as we're using it, lets fix it up and just make it public in the normal way.

File: features/docs/formatters/message.feature (hunk 2 above)
Comment (luke-hill) [reply features/docs/formatters/message.feature:70]: In the viewing of the PR it says there isn't a newline (At the end of the entire file). Just a common convention

File: lib/cucumber/runtime.rb (hunk 3 above)
Comment (aurelien-reeves) [reply lib/cucumber/runtime.rb:82]: Got it, thanks 👌

**Review by luke-hill — APPROVED** [review]

**Pull request merged**